packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -146,16 +156,33 @@ export default class DatagridController {
             this.ouiDatagridColumnBuilder.build(this.columnElements, this.getParentScope());
 
         if (this.actionColumnElements.length) {
-            builtColumns.columns.push(this.ouiDatagridColumnBuilder.buildActionColumn(this.actionColumnElements[0]));
+            this.actionColumn = this.ouiDatagridColumnBuilder.buildActionColumn(this.actionColumnElements[0]);
             this.hasActionMenu = true;
         }
 
         if (this.extraTopElements.length) {
-            this.hasExtraTopContent = true;
             this.extraTopCompiledTemplate = this.$compile(`<div>${this.extraTopElements[0].innerHTML}</div>`);
+            this.hasExtraTopContent = true;
         }
 
-        this.columns = builtColumns.columns.filter(column => !column.hidden);
+        console.log(this.columnsParameters);
+
+        this.availableColumns = angular.copy(builtColumns.columns)
+            .map(column => { // Override default with custom columns
+                const customColumn = find(this.columnsParameters, {
+                    name: column.name
+                });
+                console.log(customColumn, column.name);
```

Comment:
And a last one here :)

Reply:
Oops

Reply:
BTW, you can easily make build fail with that on CDS if you want.